### PR TITLE
[SES-708] Use consistent dialog style

### PR DESCRIPTION
**Pull request by bemusementpark**
This PR migrates dialogs to a consistent UI and leverages code reuse

A (hopefully) comprehensive list of dialogs in the app:

1. `DeleteMediaDialog` shown via `MediaOverviewActivity`

Screen Shot 2023-07-13 at 11 38 40 am

2. `DeleteMediaPreviewDialog` shown via `MediaPreviewActivity`

Screen Shot 2023-07-13 at 11 17 58 am

3. `ConversationActivity > menu > Disappearing messages`

Screen Shot 2023-07-13 at 11 54 02 am

4. `Settings > Conversations > Blocked Contacts > unblock`

Screen Shot 2023-07-13 at 12 10 16 pm

5. `BlockedDialog` shown via send message to blocked contact.

Screen Shot 2023-07-13 at 12 08 42 pm

6. `DownloadDialog` shown via click `UntrustedAttachmentView` in a visible message .
7. `JoinOpenGroupDialog` shown via `OpenGroupInvitationView` shown in `VisibleMessage` in `ConversationActivity`
8. `LinkPreviewDialog` Shown the first time the user inputs a URL that could generate a link preview...
9. `SendSeedDialog` shown when a text-only message is sent that contains the user's seed.
10. `RationaleDialog` shown via ConversationActivity on permission for mic, among many other permission requests.
11. `SettingsDialog` shown when permission is required but it is permanently denied.
12. `ClearAllDataDialog` shown via `SettingsActivity` on `clearAllData()`
13. `ListPreferenceDialog` shown via `SettingsActivity > Notifications > Notification Content`
14. `Settings > Recovery Phrase`

Screen Shot 2023-07-13 at 12 18 35 pm

15. `showEditProfilePictureUI` via `SettingsActivity`
16.  `showWarningDialog` in `SaveAttachmentTask` via `saveAttachment` via `ConversationActivity` and others...

Screen Shot 2023-07-13 at 11 19 23 am

17. `ShareLogsDialog` via `HelpSettingsActivity#shareLogs()`
18. `ConversationActivity#block` 
19. `ConversationActivity#unblock` 
20. `ConversationActivity#showGIFPicker` when `!hasSeenGIFMetaDataWarning`
21. `ConversationActivity#deleteMessages` when `recipient.isOpenGroupRecipient`
22. `ConversationActivity#deleteMessages` when `!recipient.isOpenGroupRecipient && !(allSentByCurrentUser && allHasHash)`
23. `ConversationActivity#banUser`
24. `ConversationActivity#banAndDeleteAll`
25. click control message when `(message.isCallLog && message.isFirstMissedCall)`
27. `ConversationMenuHelper#call` selected from drop down menu in `ConversationActivity`
28. `leaveClosedGroup` via `ConversationMenuHelper` via `ConversationActivity` 

Screen Shot 2023-07-13 at 11 49 59 am

29. `MuteDialog shown via `ConversationMenuHelper` via `ConversationActivity` 

Screen Shot 2023-07-13 at 11 49 22 am

30. `CallToggleListener` shown when call preference is changed to `true` to request mic permission.
31. `PrivacySettingsPreferenceFragment#setCall` triggers a dialog if `if (isEnabled && !areNotificationsEnabled(requireActivity())) // show a dialog saying that calls won't work properly if you don't have notifications on at a system level`
32. `blockConversation` via `HomeActivity`
33. `unblockConversation` via `HomeActivity`
34. `showNotifyDialog` via `HomeActivity > Long Press Message > Notification tapped`
35.  `HomeActivity#deleteConversation`
36.  `MessageRequestsActivity#onBlockConversationClick`
37.  `MessageRequestsActivity#onDeleteConversationClick`
38.  `MessageRequestsActivity#deleteAllAndBlock`
39. `PNModeActivity#register` when you sign up for Push Notifications via settings.
40.  `ConversationActivity > menu > Notifications settings`

Screen Shot 2023-07-13 at 11 39 21 am

...

Screen Shot 2023-06-06 at 8 02 09 pm
Screen Shot 2023-06-06 at 8 02 23 pm
Screen Shot 2023-06-06 at 8 02 32 pm
Screen Shot 2023-06-06 at 8 03 09 pm

Screen Shot 2023-06-06 at 8 06 19 pm
Screen Shot 2023-06-06 at 8 06 29 pm

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/permissions/RationaleDialog.kt [hunk 1]
```diff
@@ -45,6 +47,12 @@ object RationaleDialog {
             }.also(header::addView)
         }
 
+        drawables.dropLast(1).forEach {
```
Comment (hjubb) [inline app/src/main/java/org/thoughtcrime/securesms/permissions/RationaleDialog.kt]: maybe this should be behind a limit, like if there's only 3 in the it'll be `forEach` but if there's more than 3 conditionally do this "with +" logic?

Comment (bemusementpark) [reply app/src/main/java/org/thoughtcrime/securesms/permissions/RationaleDialog.kt]: > do this "with +" logic?

I don't understand.

Comment (hjubb) [reply app/src/main/java/org/thoughtcrime/securesms/permissions/RationaleDialog.kt]: nvm it's probably fine :smile:

**Review by hjubb — APPROVED** [review]

**Comment by burtonemily** [comment]
Looks good

**Pull request merged**